src/main/java/org/folio/services/item/ItemService.java

```diff
@@ -176,4 +185,27 @@ public ItemService(Context vertxContext, Map<String, String> okapiHeaders) {
   private Function<SQLConnection, Future<RowSet<Row>>> updateSingleItemBatchFactory(Item item) {
     return connection -> itemRepository.update(connection, item.getId(), item);
   }
+
+  private Future<Item> refuseWhenStatisticalCodeIdDoesNotExist(final Item item) {
+    if (item.getStatisticalCodeIds().isEmpty()) {
+      return succeededFuture(item);
+    }
+
+    return statisticalCodeRepository.getById(item.getStatisticalCodeIds())
```

Comment:
All other foreign keys are implemented using an PostgreSQL trigger.
The array can be checked with a trigger like this:
```
SET search_path TO diku_mod_inventory_storage;

CREATE OR REPLACE FUNCTION check_statistical_code_references()
RETURNS TRIGGER AS $$
DECLARE
  invalid text;
BEGIN
  SELECT ref
  INTO invalid
  FROM jsonb_array_elements_text(NEW.jsonb->'statisticalCodeIds') ref
  LEFT JOIN statistical_code ON id=ref::uuid
  WHERE id IS NULL
  LIMIT 1;
  IF FOUND THEN
    RAISE foreign_key_violation USING
      MESSAGE='statistical code doesn''t exist: ' || invalid,
      DETAIL='foreign key violation in statisticalCodeIds array of ' || TG_TABLE_NAME || ' with id=' || NEW.id,
      SCHEMA=TG_TABLE_SCHEMA,
      TABLE=TG_TABLE_NAME;
  END IF;
  RETURN NEW;
END;
$$ LANGUAGE plpgsql;

CREATE TRIGGER check_statistical_code_references_on_insert
  BEFORE INSERT ON item
  FOR EACH ROW
  WHEN (NEW.jsonb->'statisticalCodeIds' IS NOT NULL AND NEW.jsonb->'statisticalCodeIds' <> '[]')
  EXECUTE FUNCTION check_statistical_code_references();

CREATE TRIGGER check_statistical_code_references_on_update
  BEFORE UPDATE ON item
  FOR EACH ROW
  WHEN (NEW.jsonb->'statisticalCodeIds' IS NOT NULL AND NEW.jsonb->'statisticalCodeIds' <> '[]'
             AND OLD.jsonb->'statisticalCodeIds' IS DISTINCT FROM NEW.jsonb->'statisticalCodeIds')
  EXECUTE FUNCTION check_statistical_code_references();
```

What are the advantages of this Java-based code that requires yet another database query round-trip?

Reply:
Why does RMB not do that?

Reply:
The advantage is to not subject module development to implement database triggers for every case where the "framework" does not support.

I am not in favor of application layer solution and will gladly close this PR for your solution.

Reply:
What are the advantages of a database trigger over a well designed database?

Reply:
I suppose that argument could be the same as not require module development to implement application layer validation for every case the "framework" does not support the database constraint.

Anyhow, I opted for this to hopefully reduce future technical debt and additional requirements to deliver feature. This may not be the case though.

Reply:
Moved to issue, removed application layer validation, added sql script above and update schema.json.